'stmt' adds symbols too early

The following doesn't produce a Nim error, but a C error as due to the 'body: stmt' body is sem'checked twice. 'body: untyped' produces a proper Nim error.

``` nim

import os

template cd*(dir: string, body: stmt) =
  ## Sets the current dir to ``dir``, executes ``body`` and restores the
  ## previous working dir.
  let lastDir = getCurrentDir()
  setCurrentDir(dir)
  try:
    body
  finally:
    setCurrentDir(lastDir)

proc execProcess(cmd: string): string = discard

proc test =
  cd "foo":
    var output = execProcess("git tag")
  if output.len > 0:
    echo "I feel sorry"

test()
```

minimized:
```
template foo*(body: typed) =
  if true:
    body

proc test =
  foo:
    var output = ""
  echo output.len

test()
```

Error: execution of an external compiler program 'clang -c  -w  -I/Users/timothee/git_clone/nim/Nim/lib -o /tmp/nim/nimcache/timn_issue_3482.c.o /tmp/nim/nimcache/timn_issue_3482.c' failed with exit code: 1

/tmp/nim/nimcache/timn_issue_3482.c:103:24: error: use of undeclared identifier 'output'
        T5_[0] = nimIntToStr((output ? output->Sup.len : 0));
                              ^
/tmp/nim/nimcache/timn_issue_3482.c:103:33: error: use of undeclared identifier 'output'
        T5_[0] = nimIntToStr((output ? output->Sup.len : 0));

--- a/tests/misc/t3482.nim
+++ b/tests/misc/t3482.nim
@@ -1,0 +1,15 @@
+discard """
+  action: reject
+  nimout: "t3482.nim(13, 8) Error: undeclared identifier: 'output'"
+"""
+# bug #3482 (correct behavior since 1.4.0, cgen error in 1.2.0)
+template foo*(body: typed) =
+  if true:
+    body
+
+proc test =
+  foo:
+    var output = ""
+  echo output.len
+
+test()

